### PR TITLE
abci: fix abci evidence types

### PR DESCRIPTION
**Pull request by cmwaters**
## Description

Some of the evidence types weren't updated to match the changes. Such updates included:

- Removing mock evidence (we don't use that anymore)

- Adding amnesia evidence

- Removing potential amnesia evidence (this should never be committed on the chain)

### CODE REVIEW

**Comment by cmwaters** [comment]
This will require a changelog right?

**Comment by marbar3778** [comment]
> This will require a changelog right?

yes and a section in the upgrading.md file.

File: types/protobuf.go [hunk 1]
```diff
@@ -16,11 +16,10 @@ import (
 // Use strings to distinguish types in ABCI messages
 
 const (
-	ABCIEvidenceTypeDuplicateVote    = "duplicate/vote"
-	ABCIEvidenceTypePhantom          = "phantom"
-	ABCIEvidenceTypeLunatic          = "lunatic"
-	ABCIEvidenceTypePotentialAmnesia = "potential_amnesia"
-	ABCIEvidenceTypeMock             = "mock/evidence"
+	ABCIEvidenceTypeDuplicateVote = "duplicate/vote"
+	ABCIEvidenceTypePhantom       = "phantom"
```
Comment (melekes) [inline types/protobuf.go:20]: I thought we've decided to get rid of phantom evidence, no?

Comment (cmwaters) [reply types/protobuf.go:20]: Yes we have decided to get rid of it, I just wanted to do it in another PR/commit

**Pull request merged**